cryptoconditions/condition.py

```diff
@@ -232,87 +314,86 @@ def serialize_binary(self):
         CONDITION =
             VARUINT TYPE_BITMASK
             VARBYTES HASH
-            VARUINT MAX_FULFILLMENT_LENGTH
+            VARUINT MAX_COST
 
         Return:
             Serialized condition
         """
-        writer = Writer()
-        writer.write_uint16(self.type_id)
-        writer.write_var_uint(self.bitmask)
-        writer.write_var_octet_string(self.hash)
-        writer.write_var_uint(self.max_fulfillment_length)
-        return writer.buffer
-
-    def parse_binary(self, reader):
-        """
-        Parse any condition in binary format.
-
-        Will populate the condition object with data from the provided binary
-        stream.
-
-        Args:
-             reader (Reader): Binary stream containing the condition.
-        """
-        self.type_id = reader.read_uint16()
-        self.bitmask = reader.read_var_uint()
-
-        # TODO: Ensure bitmask is supported?
-        self.hash = reader.read_var_octet_string()
-        self.max_fulfillment_length = reader.read_var_uint()
-
-    def to_dict(self):
-        """
-        Generate a dict of the condition
-
-        Returns:
-            dict: representing the condition
-        """
-        return {
-            'type': 'condition',
-            'type_id': self.type_id,
-            'bitmask': self.bitmask,
-            'hash': base58.b58encode(self.hash),
-            'max_fulfillment_length': self.max_fulfillment_length
+        asn1_dict = self.to_asn1_dict()
+        asn1_condition = nat_decode(asn1_dict, asn1Spec=Asn1Condition())
+        binary_condition = der_encode(asn1_condition)
+        return binary_condition
+
+    def to_asn1_dict(self):
+        condition_type = TypeRegistry.find_by_type_id(self.type_id)
+        condition_class = condition_type['class']
+        payload = {'fingerprint': self.hash, 'cost': self.cost}
+        if condition_class.TYPE_CATEGORY == 'compound':
+            subtype_ids = [
+                TypeRegistry.find_by_name(subtype)['type_id']
+                for subtype in self.subtypes
+            ]
+            bits = ['0' for bit in range(5)]
+            for subtype_id in subtype_ids:
+                bits[subtype_id] = '1'
+            bitstring = ''.join(bits).rstrip('0')
+            payload['subtypes'] = bitstring
+        return {condition_class.TYPE_ASN1: payload}
+
+    # TODO ILP Clarification NEEDED
+    # The asn1 json payload that ILP uses in the JS implementation differs from
+    # the dictionary payload generated by pyasn1 -- hence the question:
+    # Is there a standard JSON representation of a condition?
+    def to_asn1_json(self):
```

Comment:
addressed in https://github.com/bigchaindb/cryptoconditions/pull/67/commits/0359c52e120996639760ebe0bdf977f88a9e6dc6